Bottom: 4de18263aa5dae2b8c25fbe7f164cf0cd157e689
Top:    407f5d885c4d8941ed1e3a1c8b03061d7e8d7ac9
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-05-17 10:31:02 +0800

Refresh of tuntap-introduce-tx-skb-ring

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 6001ece..be1f586 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -930,7 +930,7 @@ static netdev_tx_t tun_net_xmit(struct sk_buff *skb, struct net_device *dev)
 		spin_lock_irqsave(&tfile->wlock, flags);
 
 		head = tfile->head;
-		tail = ACCESS_ONCE(tfile->tail);
+		tail = smp_load_acquire(tfile->tail);
 
 		if (CIRC_SPACE(head, tail, TUN_RING_SIZE) >= 1) {
 			struct tun_desc *desc = &tfile->tx_descs[head];
@@ -1569,7 +1569,7 @@ static ssize_t tun_do_read(struct tun_struct *tun, struct tun_file *tfile,
 		struct tun_desc *desc;
 
 		spin_lock(&tfile->rlock);
-		head = ACCESS_ONCE(tfile->head);
+		head = smp_load_acquire(tfile->head);
 		tail = tfile->tail;
 
 		if (CIRC_CNT(head, tail, TUN_RING_SIZE) >= 1) {
